cloud/store/schema.go

```diff
@@ -355,30 +364,32 @@ func (s *schema) ShardReplicas(class, shard string) ([]string, error) {
 
 // TenantShard returns shard name for the provided tenant and its activity status
 func (s *schema) TenantShard(class, tenant string) (string, string) {
-	s.RLock()
-	defer s.RUnlock()
-
-	i := s.Classes[class]
-	if i == nil || !i.Sharding.PartitioningEnabled {
-		return "", ""
+	var exTenant, status string
```

Comment:
the retry mechanism would be called at least once, if the data already there it won't affect the performance because it won't be retried any way

Reply:
The code logic appears to be: `determineObject -> Retry::TenantShard -> ReadMetaClass -> Retry::ReadClass`. As you can see, the second retry is embedded inside the first one. It would be better to retry the entire operation just once

Reply:
i've updated the code and removed retrying `TenantShard`, it's resolved with a different comment